Query/Event/PreFilterConditionCreationEvent.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Netdudes\DataSourceryBundle\Query\Event;
+
+use Netdudes\DataSourceryBundle\DataType\DataTypeInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+class PreFilterConditionCreationEvent extends Event
+{
+    /**
+     * @var DataTypeInterface
+     */
+    private $dataType;
+
+    /**
+     * @var mixed
+     */
+    private $value;
+
+    /**
+     * @param mixed             $value
+     * @param DataTypeInterface $dataType
+     */
+    public function __construct($value, DataTypeInterface $dataType)
```

Comment:
I think this feature is not needed for now. We creating here a filter which is based on the value of the user input. Later on when we retrieving the filter we want to get this exact value back. 
The only reason for now why I created these event is that the user presented (and searched) value could be different to the value in the database and we need a possibility to provide a methodology to transform the value into the database value. 
At the moment I cannot see a use case to allow the manipulation of the input value after submitting.